### ISSUE TITLE
commentFormatting complains about some directives

### ISSUE THREAD

**Issue by AlekSi** [go-critic/go-critic#1027]
```go
//go-sumtype:decl Data
//go:noinline
```

The first one is used for https://github.com/BurntSushi/go-sumtype.

See also https://github.com/mvdan/gofumpt/issues/109 and https://github.com/mvdan/gofumpt/blob/d7cd0a5d35ce5779a01c803173a92769e330c2d1/format/format.go#L213-L225

**Comment by quasilyte** [comment]
This issue turned out to be more interesting than it looks on the surface.
I assume that the problem is inside the comment traversing code.
I'll try to fix it.

**Comment by quasilyte** [comment]
No, it's me being fooled by the `line` directive.

We had `//line` directives inside a test file, so when Go loaded a file, it assumed that we're overriding a file name (and the line). This led to a lot of confusion with warnings.

I moved `//line` directive tests to the other file.

And to fix the `go-sumtype:decl Data` case, I've added `-` the allowed tag symbols:
```diff
- ^//\w+:.*$
+ ^//[\w-]+:.*$
```
